app/bundles/PageBundle/Controller/PublicController.php

```diff
@@ -316,59 +318,54 @@ function ($a, $b): int {
     }
 
     /**
-     * @return Response|\Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+     * @return mixed[]|JsonResponse|RedirectResponse|Response
      *
-     * @throws \Exception
-     * @throws \Mautic\CoreBundle\Exception\FileNotFoundException
+     * @throws FileNotFoundException
      */
     public function previewAction(Request $request, PageConfig $pageConfig, CorePermissions $security, AnalyticsHelper $analyticsHelper, AssetsHelper $assetsHelper, ThemeHelper $themeHelper, int $id, string $objectType = null)
     {
-        $model        = $this->getModel('page');
-        $entity       = $model->getEntity($id);
-        $draftEnabled = $pageConfig->isDraftEnabled();
-        $contactId    = (int) $request->query->get('contactId');
-
-        if ($contactId) {
-            /** @var LeadModel $leadModel */
-            $leadModel = $this->getModel('lead.lead');
-            /** @var Lead $contact */
-            $contact = $leadModel->getEntity($contactId);
-        }
-
         /** @var PageModel $model */
         $model = $this->getModel('page');
         /** @var Page $page */
         $page = $model->getEntity($id);
 
-        if (!$page->getId()) {
+        if (!$page || !$page->getId()) {
             return $this->notFound();
         }
 
-        $analytics = $analyticsHelper->getCode();
+        $contactId = (int) $request->query->get('contactId');
+        if ($contactId) {
+            /** @var LeadModel $leadModel */
+            $leadModel = $this->getModel('lead.lead');
+            /** @var Lead $contact */
+            $contact = $leadModel->getEntity($contactId);
```

Comment:
The model method returns the right type. This is wrong, it can return also null
```suggestion
            $contact = $leadModel->getEntity($contactId);
```